Description: Fix menu activation under Unity global menu bar
 virt-manager disables gtk-enable-mnemonics when a running VM window is
 focused and re-enables it when the focus is lost. So clicking the global menu
 steals the focus away, changing the gtk-enable-mnemonics setting, which
 updates the menus (adding back their mnemonics). Since menu updates are done
 by removing and re-adding the same menu, the old menu is opened but
 immediately replaced by a new unopened menu...
 .
 This patch prevents this from happening under Unity. Instead, we simply
 disable mnemonics unless they are enabled in the preferences.
Author: William Hua <william.hua@canonical.com>
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1291370
Updated: 2016-01-28

Index: virt-manager-1.4.1/virtManager/console.py
===================================================================
--- virt-manager-1.4.1.orig/virtManager/console.py	2017-03-09 18:11:13.356980125 +0100
+++ virt-manager-1.4.1/virtManager/console.py	2017-03-09 18:11:13.352980162 +0100
@@ -20,6 +20,7 @@
 #
 
 import logging
+import os
 
 from gi.repository import Gtk
 from gi.repository import Gdk
@@ -175,6 +176,9 @@
             self.config.on_console_accels_changed(self._refresh_enable_accel))
 
 
+    def is_unity(self):
+        return os.getenv('XDG_CURRENT_DESKTOP') == 'Unity'
+
     def is_visible(self):
         if self.topwin:
             return self.topwin.get_visible()
@@ -783,7 +787,7 @@
             self._enable_modifiers()
         elif self._someone_has_focus():
             self._disable_modifiers()
-        else:
+        elif not self.is_unity():
             self._enable_modifiers()
 
     def _viewer_auth_rejected(self, ignore, errmsg):
